File: app/src/main/java/vn/loitp/app/activity/customviews/button/lbutton/LButtonActivity.java
package vn.loitp.app.activity.customviews.button.lbutton;

import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;

import com.core.base.BaseFontActivity;

import loitp.basemaster.R;
import vn.loitp.views.LToast;
import vn.loitp.views.button.lbutton.LButton;

public class LButtonActivity extends BaseFontActivity implements OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LButton bt0 = (LButton) findViewById(R.id.bt_0);
        bt0.setOnClickListener(this);

        LButton bt1 = (LButton) findViewById(R.id.bt_1);
        bt1.setPressedDrawable(R.drawable.circle_color_primary);
        bt1.setOnClickListener(this);

        LButton bt2 = (LButton) findViewById(R.id.bt_2);
        bt2.setPressedDrawable(R.drawable.bt_color_primary);
        bt2.setOnClickListener(this);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_l_button;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.bt_0:
            case R.id.bt_1:
            case R.id.bt_2:
                LToast.INSTANCE.show(activity, "Click");
                break;
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/button/lbutton/LButtonActivity.kt
package vn.loitp.app.activity.customviews.button.lbutton

import android.os.Bundle
import android.view.View
import android.view.View.OnClickListener
import androidx.core.view.isVisible
import com.annotation.IsFullScreen
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_button_l.*
import vn.loitp.app.R

@LogTag("LButtonActivity")
@IsFullScreen(false)
class LButtonActivity : BaseFontActivity(), OnClickListener {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_button_l
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.viewShadow?.isVisible = true
            this.tvTitle?.text = LButtonActivity::class.java.simpleName
        }

        bt0.setOnClickListener(this)

        bt1.setPressedDrawable(R.drawable.l_circle_color_primary)
        bt1.setOnClickListener(this)

        bt2.setPressedDrawable(R.drawable.l_bt_color_primary)
        bt2.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v) {
            bt0, bt1, bt2 ->
                showShortInformation(msg = getString(R.string.click))
        }
    }
}
